src/shims/unix/linux/fs.rs

```diff
@@ -0,0 +1,167 @@
+use rustc_middle::ty::ScalarInt;
+
+use crate::{machine::SIGRTMAX, *};
+use epoll::{Epoll, EpollEvent};
+use event::Event;
+use socketpair::SocketPair;
+
+use shims::unix::fs::EvalContextExtPrivate;
+
+pub mod epoll;
+pub mod event;
+pub mod socketpair;
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriInterpCx<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
+    fn epoll_create1(&mut self, flags: &OpTy<'tcx, Provenance>) -> InterpResult<'tcx, i32> {
```

Comment:
Please make this function (and the others in this file) return a `Scalar<Provenance>` rather than an `i32`. (That's the style we have been using for all new functions, and I ported some but not all of the existing ones.)